core/services/ocr2/plugins/threshold/decryption_queue_test.go

```diff
@@ -0,0 +1,357 @@
+package decryption_queue
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestNewThresholdDecryptor(t *testing.T) {
+	dq := NewThresholdDecryptor(5, 1000, 1000)
+	if dq.maxQueueLength != 5 || dq.maxCiphertextBytes != 1000 || dq.completedRequestsCacheTimeoutMs != 1000 {
+		t.Error("Init failed to set maxQueueSize and localCacheTimeoutMs")
+	}
+}
+
+func TestDecryptAndResultReady(t *testing.T) {
+	dq := NewThresholdDecryptor(5, 1000, 1000)
+
+	go func() {
+		waitForPendingRequestToBeAdded(t, dq, []byte("1"))
+		dq.ResultReady([]byte("1"), []byte("decrypted"))
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(1000)*time.Millisecond)
+	defer cancel()
+
+	pt, err := dq.Decrypt(ctx, []byte("1"), []byte("encrypted"))
+	if err != nil {
+		t.Error("Unexpected error:", err)
+	}
+	if !reflect.DeepEqual(pt, []byte("decrypted")) {
+		t.Error("Decrypt did not return the expected plaintext")
+	}
+}
+
+func TestCiphertextTooLarge(t *testing.T) {
```

Comment:
Method focused tests are typically named like `Test<type>_<method>`:
```suggestion
func Test_decryptionQueue_Decrypt_CiphertextTooLarge(t *testing.T) {
```
The common prefixes can optionally be grouped together as well:
```suggestion
func Test_decryptionQueue_Decrypt(t *testing.T) {
	t.Run("CiphertextTooLarge", func(t *testing.T) {
```
Then the full sub-test name would be: `Test_decryptionQueue_Decrypt/CiphertextTooLarge`

Reply:
@jmank88 Thank you for the review! I have updated the names accordingly